Client - check for duplicate email address before new user registration

clairecatohanson opened the pull request:
This PR adds functionality to check whether an email address is already in the database before registering a new user to ensure that each user in the system has a unique email address. If n email address already exists, the user will receive a window alert indicating that the email address is already taken, and the user will not be posted to the database.

Addresses #48 

To Test:
1. Start up the api in the debugger, and run npm start while checked out on this branch of the client repo.
2. Navigate to localhost:3000/register in your browser.
3. Fill out all fields of the form with a test user details and click Register. Verify that the user was added to your version of db.sqlite3. 
4. Go back to localhost:3000/register and try to register a new user, using the same email address that was used in step 3. Click Register. You should receive a window alert pop-up with an error indicating that the email address has already been registered.

evymai reviewed (changes requested):
When inputting a registering a new non-existing user, "Your email has already been registered" still appears.

clairecatohanson commented:
> When inputting a registering a new non-existing user, "Your email has already been registered" still appears.

I believe I fixed the error. There was a key naming error on line 54 that I addressed in my most recent push.

evymai reviewed (approved):
That change fixed the problem. Looks good!
